js/src/tests/evaluate_attachments.int.test.ts

```diff
@@ -0,0 +1,469 @@
+
+  // Verify results
+  expect(evalRes.results).toHaveLength(2);
+
+  for (const result of evalRes.results) {
+    // NOTE: THIS DOES NOT WORK YET BECAUSE THE ATTACHMENTS ARE NOT TRACED CORRECTLY
+    // THIS SHOULD BE FIXED ASAP
+    // expect(result.run.attachments).toBeDefined();
+    expect(result.run).toBeDefined();
+    expect(result.example).toBeDefined();
+    expect(result.evaluationResults).toBeDefined();
+
+    // Verify evaluator results
+    const evalResults = result.evaluationResults.results;
+    expect(evalResults).toHaveLength(1); // Should have attachment_presence and attachment_processing results
+
+    // Check that attachments were properly processed
+    const evalResult = evalResults.find((r) => r.key === "key");
+    expect(evalResult?.score).toBe(1);
+  }
+
+  // Cleanup
+  await client.deleteDataset({ datasetName });
+});
+
+test("evaluate with attachments not in target function", async () => {
+  const client = new Client();
+  const datasetName = `test_dataset_attachments_${uuidv4()}`;
+  const dataset = await client.createDataset(datasetName);
+
+  // Create examples with attachments
+  const example: ExampleUploadWithAttachments = {
+    inputs: { question: "What is shown in the image?" },
+    outputs: { answer: "test image" },
+    attachments: {
+      image: ["image/png", Buffer.from("fake image data for testing")],
+    },
+  };
+
+  await client.uploadExamplesMultipart(dataset.id, [example]);
+
+  // Define target function that processes attachments
+  const targetFunc = async (_inputs: Record<string, any>) => {
+    return { answer: "test image" };
+  };
+
+  const customEvaluator = async ({ attachments }: { attachments?: any }) => {
+    expect(attachments).toBeDefined();
+    expect(attachments?.["attachment.image"]).toBeDefined();
+    const expectedData = new Uint8Array(
+      Buffer.from("fake image data for testing")
+    );
+    const attachmentData: Uint8Array | undefined = attachments?.[
+      "attachment.image"
+    ].presigned_url
+      ? new Uint8Array(
+          (await fetch(attachments?.["attachment.image"].presigned_url).then(
+            (res) => res.arrayBuffer()
+          )) as ArrayBuffer
+        )
+      : undefined;
+    if (!arraysEqual(attachmentData ?? new Uint8Array(), expectedData)) {
+      throw new Error("Image data does not match expected data");
+    }
+    return {
+      key: "key",
+      score: 1,
+    };
+  };
+
+  // Run evaluation
+  const evalRes = await evaluate(targetFunc, {
+    data: datasetName,
+    description: "Testing attachment handling in evaluation",
+    client: client,
+    evaluators: [customEvaluator],
+    numRepetitions: 2,
+    includeAttachments: true,
+  });
+
+  // Verify results
+  expect(evalRes.results).toHaveLength(2);
+
+  for (const result of evalRes.results) {
+    // NOTE: THIS DOES NOT WORK YET BECAUSE THE ATTACHMENTS ARE NOT TRACED CORRECTLY
+    // THIS SHOULD BE FIXED ASAP
+    // expect(result.run.attachments).toBeDefined();
```

Comment:
maybe just mark this as a todo rather than an urgent fix? It's not breaking anything just not supported yet